dali/kernels/imgproc/convolution/separable_convolution_gpu.h

```diff
@@ -146,14 +149,15 @@ struct SeparableConvolutionGpu<Out, In, W, 3, has_channels, is_sequence> {
   static constexpr int channel_axes = static_cast<int>(has_channels);
   static constexpr int ndim = sequence_axes + axes + channel_axes;
   using Intermediate = decltype(std::declval<W>() * std::declval<In>());
-  static constexpr bool kUseOutAsIntermediate = sizeof(Intermediate) == sizeof(Out);
+  static constexpr bool outFitsIntermediate = sizeof(Intermediate) == sizeof(Out);
```

Comment:
Yeah, it should work I guess, but I am just reusing existing code here and `reinterpret` won't work for that, so it would possibly require some extra functionality that goes sample by sample to create new TensorListView. How about making it a follow up small task?

Reply:
Sure